app/src/main/java/com/infomaniak/mail/utils/MessageBodyUtils.kt

```diff
@@ -39,77 +40,97 @@ object MessageBodyUtils {
+    private fun findFirstKnownParentQuoteDescriptor(htmlDocumentWithoutQuote: Document): String {
         var currentQuoteDescriptor = ""
         for (quoteDescriptor in quoteDescriptors) {
-            val quotedContentElement = htmlDocumentWithoutQuote.select(quoteDescriptor).first()
-            if (quotedContentElement != null) {
-                htmlQuotes.add(quotedContentElement)
+            val quotedContentElement = htmlDocumentWithoutQuote.selectElementAndFollowingSiblings(quoteDescriptor)
+            if (quotedContentElement.isNotEmpty()) {
                 quotedContentElement.remove()
                 currentQuoteDescriptor = quoteDescriptor
-                break
             }
         }
         return currentQuoteDescriptor
     }
 
     private fun splitBodyAndQuote(
-        htmlQuotes: MutableList<Element>,
         htmlDocumentWithQuote: Document,
         currentQuoteDescriptor: String,
+        blockquoteElement: Element?,
     ): Pair<String, String?> {
-        return htmlQuotes.lastOrNull()?.let { htmlQuote ->
-            val quotedContentElements = htmlDocumentWithQuote.select(currentQuoteDescriptor)
-            if (currentQuoteDescriptor == blockquote) {
-                for (quoteElement in quotedContentElements) {
-                    if (quoteElement.toString() == htmlQuote.toString()) {
-                        quoteElement.remove()
+        return when {
+            currentQuoteDescriptor == blockquote -> {
+                for (quotedContentElement in htmlDocumentWithQuote.select(currentQuoteDescriptor)) {
+                    if (quotedContentElement.toString() == blockquoteElement.toString()) {
+                        quotedContentElement.remove()
                         break
                     }
                 }
-                htmlDocumentWithQuote.toString() to htmlQuote.toString()
-            } else {
-                val firstQuotedContent = quotedContentElements.first()
-                firstQuotedContent?.remove()
-                htmlDocumentWithQuote.toString() to firstQuotedContent.toString()
+                htmlDocumentWithQuote.toString() to blockquoteElement.toString()
+            }
+            currentQuoteDescriptor.isNotEmpty() -> {
+                val quotedContentElements = htmlDocumentWithQuote.selectElementAndFollowingSiblings(currentQuoteDescriptor)
+                quotedContentElements.remove()
+                htmlDocumentWithQuote.toString() to quotedContentElements.toString()
+            }
+            else -> {
+                htmlDocumentWithQuote.toString() to null
             }
-        } ?: (htmlDocumentWithQuote.toString() to null)
+        }
+    }
+
+    //region Utils
+    /**
+     * Some email clients rename css classes to prefix them for example.
+     * We match all the css classes that contain the quote, in case this one has been renamed.
+     * @return a new css query
```

Comment:
```suggestion
     * Some Email clients rename CSS classes to prefix them.
     * We match all the CSS classes that contain the quote, in case this one has been renamed.
     * @return a new CSS query
```